lib/rex/proto/ldap/server.rb

```diff
@@ -149,104 +155,177 @@ def dispatch_request(cli, data)
+                    processed_pdu_data = @auth_provider.process_login_request(user_login)
+                    processed_pdu_data = processed_pdu_data.merge(state[client])
+                    if processed_pdu_data[:result_code] == Net::LDAP::ResultCodeSaslBindInProgress
+                      server_creds = processed_pdu_data[:server_creds]
+                      context_code = 7
+                    else
+                      processed_pdu_data[:result_message] = "LDAP Login Attempt => From:#{processed_pdu_data[:name]}\n Username:#{processed_pdu_data[:user]}\n #{processed_pdu_data[:private_type]}:#{processed_pdu_data[:private]}\n"
+                      processed_pdu_data[:result_message] += " Domain:#{processed_pdu_data[:domain]}" if processed_pdu_data[:domain]
+                      post_pdu = true
+                    end
+                    processed_pdu_data[:pdu_type] = pdu.app_tag
+                    encode_ldap_response(
+                      pdu.message_id,
+                      processed_pdu_data[:result_code],
+                      '',
+                      Net::LDAP::ResultStrings[processed_pdu_data[:result_code]],
+                      Net::LDAP::PDU::BindResult,
+                      server_creds,
+                      context_code
+                    )
+                  when Net::LDAP::PDU::SearchRequest
+                    filter = Net::LDAP::Filter.parse_ldap_filter(pdu.search_parameters[:filter])
+                    attrs = pdu.search_parameters[:attributes].empty? ? :all : pdu.search_parameters[:attributes]
+                    res = search_result(filter, pdu.message_id, attrs)
+                    if res.nil? || res.empty?
+                      result_code = Net::LDAP::ResultCodeNoSuchObject
+                    else
+                      client.write(res)
+                      result_code = Net::LDAP::ResultCodeSuccess
+                    end
+                    processed_pdu_data[:pdu_type] = pdu.app_tag
+                    encode_ldap_response(
+                      pdu.message_id,
+                      result_code,
+                      '',
+                      Net::LDAP::ResultStrings[result_code],
+                      Net::LDAP::PDU::SearchResult
+                    )
+                  when Net::LDAP::PDU::UnbindRequest
+                    client.close
+                    nil
+                  else
+                    if suitable_response(pdu.app_tag)
+                      result_code = Net::LDAP::ResultCodeUnwillingToPerform
+                      encode_ldap_response(
+                        pdu.message_id,
+                        result_code,
+                        '',
+                        Net::LDAP::ResultStrings[result_code],
+                        suitable_response(pdu.app_tag)
+                      )
+                    else
+                      client.close
+                    end
+                  end
+
+            if @pdu_process[pdu.app_tag] && post_pdu
```

Comment:
I thought since post processing of `pdu` is not to be done in some cases like sending type2 response even when `processed_pdu_data` is not empty, the variable can be used to determine if further processing of data is required or just send the response.

Reply:
I see the flow you are focused on and it can work for now, this adjustment can be considered optional.  

The decision on how the data should be processed could be left up to the module author, consider a theoretical module that for some reason gathers how often a type2 is sent and no further interaction occurs from a target. The suggestion would allow the service to be more flexible by offering modules the opportunity to have more information about what requests the service is handling.